cpp/src/slice2cs/Gen.cpp

```diff
@@ -3089,14 +3104,14 @@ Slice::Gen::ImplVisitor::visitOperation(const OperationPtr& op)
     string ns = getNamespace(cl);
     string opName = operationName(op);
 
-    list<ParamInfo> outParams = getAllOutParams(op);
+    list<ParamInfo> outParams = getAllOutParams(op, Direction::Outgoing);
```

Comment:
Maybe we can add a few helper methods rather than explicitly pass a direction 
```
getAllOutgoingOutParams
getAllOutgoingInParams
getAllIncomingOutParams
getAllIncomingInParams
```